src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -107,21 +108,23 @@ public void runAppCommand(List<String> args) throws ProcessRunnerException {
     command.addAll(args);
 
     command.add("--quiet");
-    command.addAll(GcloudArgs.get("credential-file-override", appCommandCredentialFile));
+
+    Map<String, String> environment = Maps.newHashMap();
+
+    if (appCommandCredentialFile != null) {
```

Comment:
Minor, but the flow of this might be better if it was moved next to the other environment setting logic (without format in between).